mahjong/record/utils/event.py

```diff
@@ -112,3 +113,99 @@ def __str__(self):
+                     'score': attrs['ten'].split(',')[2] + '00'}
+            dora = {'event_type': 'DORA ', 'player': '0',
+                    'player_see': str(tile_from_tenhou(int(attrs['seed'].split(',')[-1]))),
+                    'player_show': str(tile_from_tenhou(int(attrs['seed'].split(',')[-1])))}
+            initround = {'event_type': 'ROUND', 'score': attrs['seed'].split(',')[2] + '000'}
+            if ('hai3' in attrs.keys()) and (attrs['hai3'] != ''):
+                dict3 = {'event_type': 'INIT4', 'player': '3', 'player_see': ''.join(
+                    [str(tile_from_tenhou(i)) for i in sorted([int(i) for i in attrs['hai3'].split(',')])]),
+                         'score': attrs['ten'].split(',')[3] + '00'}
+                return [dict0, dict1, dict2, dict3, dora, initround]
+            else:
+                return [dict0, dict1, dict2, dora, initround]
+        draw = draw_tile_change(event)
+        if draw:
+            player, tile = draw['player'], draw['tile']
+            return [{'event_type': 'DRAW ', 'player': player, 'player_see': str(tile)}]
+        discard = discard_tile_change(event)
+        if discard:
+            player, tile = discard['player'], discard['tile']
+            return [{'event_type': 'DISCA', 'player': player, 'player_show': str(tile)}]
+        if is_open_hand(event):
+
+            player = attrs['who']
+            item = meld_from(event)
+            type_of = unpack_with(MeldTypeData, meld_type_unpacker, event.attrib['m'])
+            source = item.from_who
+            self_tile = item.self_tiles
+            borrow_tile = item.borrowed_tiles
+            if type_of.syuntsu:
+                return [{'event_type': 'CHI  ', 'player': player,
+                         'player_show': ''.join([str(tile_from_tenhou(int(i))) for i in self_tile]),
+                         'player_open': ''.join([str(tile_from_tenhou(int(i))) for i in borrow_tile]) + ''.join(
+                             [str(tile_from_tenhou(int(i))) for i in self_tile]), 'origin': source}]
+            elif type_of.koutsu:
+                return [{'event_type': 'PENG ', 'player': player,
+                         'player_show': ''.join([str(tile_from_tenhou(int(i))) for i in self_tile]),
+                         'player_open': ''.join([str(tile_from_tenhou(int(i))) for i in borrow_tile]) + ''.join(
+                             [str(tile_from_tenhou(int(i))) for i in self_tile]), 'origin': source}]
+            elif type_of.chakan:
+                return [{'event_type': 'GANG', 'player': player,
+                         'player_show': ''.join([str(tile_from_tenhou(int(i))) for i in self_tile]),
+                         'player_open': ''.join([str(tile_from_tenhou(int(i))) for i in borrow_tile]) + ''.join(
+                             [str(tile_from_tenhou(int(i))) for i in self_tile]), 'origin': source}]
+            elif type_of.nuki:
+                return [{'event_type': 'KITA ', 'player': player,
```

Comment:
`'KITA '` leaves a blanke

Reply:
No special meanings, just to make sure the length of  'event_type' is 5 in all conditions.

Reply:
a `DataFrame` will always treat string as pure python object, so no fixed length needed.